### PR TITLE
Fetch Data from YouTube API

### PR DESCRIPTION
**Pull request by Yatin-Bajaj**
Home Task:

Create page using react components as per Home task Link for Movies Webpage (stateless/stateful) (Advanced)
Add some performance tweaks (shouldComponentUpdate, Pure Component, immutable.js)

react task 3
react task 2
react task 1

### CODE REVIEW

File: src/components/Main.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useState } from "react";
+import Header from "./Header/Header";
+import Youtube from "./Youtube/Youtube";
+
+const Main = () => {
+    const [searchQuery, setSearchQuery] = useState("epam");
```
Comment (sahil1504001) [inline src/components/Main.jsx]: keep default searchQuery in const and import and use

File: src/components/UI/Card.module.css [hunk 2]
```diff
@@ -0,0 +1,68 @@
+.card {
+  color: #000;
+  background-color: #fff;
+  max-width: 300px;
+  max-height: 400px;
+  box-shadow:0 1px 3px 0 rgba(51,51,51,.2);
+  border-radius: 16px;
+  padding: 12px;
+
```
Comment (sahil1504001) [inline src/components/UI/Card.module.css]: un-necessary empty space

File: src/components/UI/CardItem.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                    </div>
+                    <div className={classes["author"]}>
+                        Author : <span>{author}</span>
+                    </div>
+                    <div className={classes["publish_date"]}>
+                        Published At : <span>{publishedAt}</span>
+                    </div>
+                    <div className={classes["view_count"]}>
+                        View Count : <span>{viewCount}</span>
+                    </div>
```
Comment (sahil1504001) [inline src/components/UI/CardItem.jsx]: move strings to const file

File: src/components/UI/CardItem.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                </div>
+            </div>
+            <div className={classes["link"]}>
+                <a
+                    href={videoUrl}
+                    className={classes["link_btn"]}
+                    target="_blank"
+                    rel="noreferrer"
+                >
+                    Link
```
Comment (sahil1504001) [inline src/components/UI/CardItem.jsx:49]: move strings to const file

File: src/components/Youtube/Youtube.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    async getYoutubeData(searchQuery) {
+        const data = await fetchData(searchQuery);
+        this.setState(() => {
+            return {
+                youtubeData: data.items,
+                loading: false,
+            };
+        });
+    }
+    async componentDidMount() {
```
Comment (sahil1504001) [inline src/components/Youtube/Youtube.jsx:27]: 1 empty line before current line

File: src/components/Youtube/Youtube.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            this.setState((prevState) => {
+                return {
+                    ...prevState,
+                    loading: true,
+                };
+            });
+            this.getYoutubeData(this.props.searchQuery);
+        }
+    }
+    render() {
```
Comment (sahil1504001) [inline src/components/Youtube/Youtube.jsx:49]: 1 empty line before current line

File: src/util/convertDateFormat.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+function convertDateFormat(currentDate = new Date()) {
+    const options = { year: "numeric", month: "long", day: "numeric" };
+    return new Date(currentDate).toLocaleDateString("en-US", options);
+}
+
+export default convertDateFormat
```
Comment (sahil1504001) [inline src/util/convertDateFormat.js]: eof missing

File: src/util/convertViewCount.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+function convertViewsCount(views = 0) {
+    if (views >= 1000000) {
+        return views.slice(0, -6) + "." + views.slice(-6, -5) + "M";
+    }
+    if (views >= 1000) {
+        return views.slice(0, -3) + "k";
+    }
+    return views;
+}
+export default convertViewsCount
```
Comment (sahil1504001) [inline src/util/convertViewCount.js]: eof missing

**Comment by Yatin-Bajaj** [comment]
All the requirements have been fulfilled and all the changes made look's good to me and mentor Sahil both.

**Pull request merged**